### ISSUE TITLE
ERROR: download_ex_doc failed.

### ISSUE THREAD

**Issue by conradwt** [erlang/otp#8656]
**Describe the bug**

When building the documentation (i.e. Building docs...), I'm getting the following error message:

```zsh
ERROR: download_ex_doc failed.
( cd /Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/erts/start_scripts/tmp && \
	  erlc -W  -I/Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/lib/kernel/ebin -I/Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/lib/stdlib/ebin -I/Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/lib/sasl/ebin +otp_build -o /Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/bin/start_sasl.script /Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/erts/start_scripts/start_sasl.rel )
/opt/local/bin/ginstall -c -d /Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/erts/start_scripts/tmp
( cd /Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/erts/start_scripts/tmp && \
	  erlc -W  -I/Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/lib/kernel/ebin -I/Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/lib/stdlib/ebin -I/Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/lib/sasl/ebin +no_warn_sasl +otp_build -o /Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/bin/start_clean.script /Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/erts/start_scripts/start_clean.rel )
/opt/local/bin/ginstall -c -d /Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/erts/start_scripts/tmp
( cd /Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/erts/start_scripts/tmp && \
	  erlc -W  -I/Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/lib/kernel/ebin -I/Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/lib/stdlib/ebin -I/Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/lib/sasl/ebin +no_warn_sasl +otp_build +no_dot_erlang -o /Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/bin/no_dot_erlang.script /Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/erts/start_scripts/no_dot_erlang.rel )
Neither sha1sum nor sha256sum found to verify /Users/conradwt/.asdf/plugins/erlang/kerl-home/builds/asdf_27.0.1/otp_src_27.0.1/bin/ex_doc
Please check manually that the correct ex_doc was downloaded.
```

**To Reproduce**

```zsh
asdf install erlang 27.0.1
```

**Expected behavior**

I expected the build to complete without error.

**Affected versions**

OTP 27.0.1

**Additional context**

I didn't experience this issue when installing OTP 27.0 on Apple M1 using macOS 14.5.

**Comment by jhogberg** [comment]
Thanks for your report! Unfortunately most of us are on vacation right now, we'll revisit this once we're back. :-)

**Comment by GRoguelon** [comment]
@conradwt I found the issue. The problem is located in [`otp_build`](https://github.com/erlang/otp/blob/OTP-27.0.1/otp_build#L1013-L1034), the build tries to check if either `sha256sum` or `sha1sum` commands are available. If not it returns the error you shared. 

I see you are on macOS  so you can fix it by install `coreutils`. 

With brew just do: 
```shell
brew install coreutils
```
After you install your Erlang version you can remove it it is used to check the integrity of `ex_doc`.

**Comment by conradwt** [comment]
@GRoguelon Great catch because I totally missed this issue.  Anyway, I'm guessing it may be best to leave these tools installed because ex_doc integrity will need to be checked for each new OTP installation >= 27.0.1.  

It's interesting that OTP team would introduce a dependency that's not called out in the documentation and didn't exist in OTP 27.0.

**Comment by GRoguelon** [comment]
@conradwt I was blocked myself on the issue.

macOS has the command `shasum` which works exactly the current implementation you just need to provide which hash function you want with the option `-a`, like: `-a 256`. Maybe it should be considered as fallback in case of miss on the `sha256sum` and `sha1sum`.

https://ss64.com/mac/shasum.html

**Comment by conradwt** [comment]
@GRoguelon Shasum and other dependencies should be documented and added to the build process.  The build process must use the appropriate command for the underlying OS here.  This is only an issue for local development because most tend not to install docs in production.

**Comment by kikofernandez** [comment]
fixed by #8707, closing issue.
